src/main/java/com/example/autokis/parser/impl/ParserImpl.java

```diff
@@ -0,0 +1,167 @@
+package com.example.autokis.parser.impl;
+
+import com.example.autokis.parser.Parser;
+import com.example.autokis.parser.model.Product;
+import lombok.extern.slf4j.Slf4j;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component("auto-elegant")
+@Slf4j
+public class ParserImpl implements Parser {
+    private static final String WEBSITE_MAIN_URL = "https://auto-elegant.com";
+    private Document currentDocument;
+    private static final String PAGE_PREFIX = "?page=";
+
+    private void connectToWebsite() throws IOException {
+        currentDocument = Jsoup.connect(WEBSITE_MAIN_URL).get();
+    }
+
+    public List<String> parseCategories() throws IOException {
```

Comment:
I would extract this logic from here and have some other class that is responsible for providing all required urls like categories and product urls.